MoltenVK/MoltenVK/GPUObjects/MVKVulkanAPIObject.mm

```diff
@@ -56,7 +56,7 @@
 
 VkResult MVKVulkanAPIObject::setDebugName(const char* pObjectName) {
 	[_debugName release];
-	_debugName = [[NSString stringWithUTF8String: pObjectName] retain];		// retained
+	_debugName = pObjectName ? [[NSString stringWithUTF8String: pObjectName] retain] : [@"" retain];
```

Comment:
Is there a reason why you added an empty default string? The design intention was to leave `_debugName` as `nil` if it is not set, for potential performance optimizations...and to allow Metal's default names to continue to appear. Note the use of `setLabelIfNotNil()` everywhere. So the appropriate behaviour here should be `? ... : nil`.

Reply:
You're right. It makes more sense to simply set it to `nil`.